### DISCUSSION TITLE
If there is no page content for a language and that language is active, in the page-tree view, that page shows as empty, editors might mistakenly delete such a page

### DISCUSSION THREAD

**Discussion by jrief** [django-cms/django-cms#7984] [General]
In Django-CMS versions>4 a dangerous and somehow unintuitive user experience has been introduced. Let me explain the problem:

You have a site with multiple languages but the content for some of the languages is not filled. When that language is selected in the page-tree view, most pages present themselves as empty:

<img width="948" alt="Screenshot 2024-09-04 at 10 43 00" src="https://github.com/user-attachments/assets/e4d6d481-d9d7-4706-9bdd-2ad7782d76bc">

Some of our users therefore falsely assumed that these pages can be deleted and sometime they did so. And by deleting a supposedly empty page, they unintentionally also deleted the German version. This of course caused major problems because they had to restart all over again.

Therefore I'd like to change the page deletion logic. When deleting a page, only the `PageContent`-object is deleted. Only if for that specific `Page`-object all content objects are of type `EmptyPageContent`, then the `Page`-object is deleted as well.

What do you think?

**Comment by fsbraun** [comment]
This PR https://github.com/django-cms/django-cms/pull/7955 does lead to fallback languages being shown instead of empty (in italics and including the language key). It should at least address part of the issue. So, this is what a mostly empty page would look like:

![image](https://github.com/user-attachments/assets/7dacfab3-1088-480a-8dd8-40e15ca6fad5)

**Comment by marksweb** [comment]
Sounds like the result of v4 being developed around a use case which didn't use the page tree.

**Comment by fsbraun** [comment]
In addition to the above usability improvement, I could imagine separating "Delete language" and "Delete page (and all languages)".

**Comment by jrief** [comment]
How about changing the layout to something like this:
<img width="1651" alt="Screenshot 2024-09-30 at 14 49 43" src="https://github.com/user-attachments/assets/5a2d2786-dd69-4c83-b793-7a4688b79a08">

In my opinion, the distinction between existing and translated pages should be clearer. Therefore, in addition to render the title of empty pages in italics, I also would render the title of existing pages in bold.

What do you think?

**Comment by fsbraun** [comment]
With #8070 also improves clarity of the delete confirmation message
